esphome/components/opentherm/hub.cpp

```diff
@@ -0,0 +1,331 @@
+  if (request_id == MessageId::STATUS) {
+    // NOLINTBEGIN
+    bool const ch_enabled = this->ch_enable &&
+#ifdef OPENTHERM_READ_ch_enable
+                            OPENTHERM_READ_ch_enable
+#else
+                            true
+#endif
+                            &&
+#ifdef OPENTHERM_READ_t_set
+                            OPENTHERM_READ_t_set > 0.0
+#else
+                            true
+#endif
+        ;
+
+    bool dhw_enabled = this->dhw_enable &&
+#ifdef OPENTHERM_READ_dhw_enable
+                       OPENTHERM_READ_dhw_enable
+#else
+                       true
+#endif
+        ;
+    bool cooling_enabled = this->cooling_enable &&
+#ifdef OPENTHERM_READ_cooling_enable
+                           OPENTHERM_READ_cooling_enable
+#else
+                           true
+#endif
+                           &&
+#ifdef OPENTHERM_READ_cooling_control
+                           OPENTHERM_READ_cooling_control > 0.0
+#else
+                           true
+#endif
+        ;
+    bool otc_enabled = this->otc_active &&
+#ifdef OPENTHERM_READ_otc_active
+                       OPENTHERM_READ_otc_active
+#else
+                       true
+#endif
+        ;
+    bool ch2_enabled = this->ch2_active &&
+#ifdef OPENTHERM_READ_ch2_active
+                       OPENTHERM_READ_ch2_active
+#else
+                       true
+#endif
+                       &&
+#ifdef OPENTHERM_READ_t_set_ch2
+                       OPENTHERM_READ_t_set_ch2 > 0.0
+#else
+                       true
+#endif
+        ;
+    // NOLINTEND
+
+    data.type = MessageType::READ_DATA;
+    data.id = MessageId::STATUS;
+    data.valueHB = ch_enabled | (dhw_enabled << 1) | (cooling_enabled << 2) | (otc_enabled << 3) | (ch2_enabled << 4);
+
+    return data;
+  }
+  return OpenthermData();
+}
+
+OpenthermHub::OpenthermHub() : Component() {}
+
+void OpenthermHub::process_response(OpenthermData &data) {
+  ESP_LOGD(OT_TAG, "Received OpenTherm response with id %d (%s)", data.id,
+           opentherm_->message_id_to_str((MessageId) data.id));
+  ESP_LOGD(OT_TAG, "%s", opentherm_->debug_data(data).c_str());
+}
+
+void OpenthermHub::setup() {
+  ESP_LOGD(OT_TAG, "Setting up OpenTherm component");
+  this->opentherm_ = new OpenTherm(this->in_pin_, this->out_pin_);  // NOLINT because hub is never deleted
+  if (!this->opentherm_->initialize()) {
+    ESP_LOGE(OT_TAG, "Failed to initialize OpenTherm protocol. See previous log messages for details.");
+    return;
+  }
+
+  // Ensure that there is at least one request, as we are required to
+  // communicate at least once every second. Sending the status request is
+  // good practice anyway.
+  this->add_repeating_message(MessageId::STATUS);
+
+  this->current_message_iterator_ = this->initial_messages_.begin();
+  initialized_ = true;
+}
+
+void OpenthermHub::on_shutdown() { this->opentherm_->stop(); }
+
+void OpenthermHub::loop() {
+  if (!initialized_)
+    return;
+
+  if (sync_mode_) {
+    sync_loop_();
```

Comment:
Please make sure **all** member variables and function calls have `this->` prefixed.
```suggestion
  if (this->sync_mode_) {
    this->sync_loop_();
```

Reply:
Done.